packages/common/container-definitions/src/loader.ts

```diff
@@ -389,21 +389,48 @@ export interface IContainer extends IEventProvider<IContainerEvents> {
 	proposeCodeDetails(codeDetails: IFluidCodeDetails): Promise<boolean>;
 
 	/**
-	 * Attaches the Container to the Container specified by the given Request.
+	 * When a container is created in Fluid it starts in what we call a detached state,
+	 * which means it is detached from server storage and unavailable for other users to open or collaborate with.
+	 * A detached container can be serialized, stored, and rehydrated by the application.
 	 *
-	 * @privateRemarks
+	 * To finish the creation of a container and make it available for other users to open and collaborate with it must be attached,
+	 * which means it is attached to or persisted in server storage. This function is what accomplishes that.
+	 *
+	 * In order to craft the IRequest parameter necessary to call this function you should use functionality provided
+	 * by the driver you are using. This is usually a free function called createCreateNewRequest, which should take
+	 * strongly typed arguments related to your driver and server.
+	 *
+	 * By default an attach failure will result in the closure of the container.
+	 * However, closure can now be avoided in most cased by setting:
+	 * Fluid.Container.RetryOnAttachFailure to true
+	 * via the config provider passed to the loader.
```

Comment:
This could use a doc-link to wherever we describe how to use the config provider.